portfolio/src/main/webapp/script.js

```diff
@@ -141,3 +142,40 @@ async function displayNickname() {
     element.innerHTML = 'Your current nickname is ' + nickname + '.';
   }
 }
+
+/**
+ * Creates a map of my favorite places in Orange County
+ */
+function createMap() {
+  const map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: 33.6, lng: -117.77},
+    zoom: 10
+  });
+
+  addLandmark(map, 33.649397, -117.742604, 'Irvine Spectrum Center', 'places/spectrum.html');
+  addLandmark(map, 33.574302, -117.840324, 'Crystal Cove State Park', 'places/crystalcove.html');
+  addLandmark(map, 33.476702, -117.720383, 'Salt Creek Beach', 'places/saltcreek.html');
+  addLandmark(map, 33.558982, -117.668982, 'The Shops at Mission Viejo', 'places/mvmall.html');
+  addLandmark(map, 33.576321, -117.726727, 'Regal Edwards Aliso Viejo & IMAX', 'places/movies.html');
+  addLandmark(map, 33.690977, -117.888958, 'South Coast Plaza', 'places/scplaza.html');
+  addLandmark(map, 33.812095, -117.918980, 'Disneyland Park', 'places/disney.html');
+  addLandmark(map, 33.637634, -117.592207, 'Mathnasium of RSM', 'places/mathnasium.html');
+  
+}
+
+/**
+ * Adds a marker that shows an info window when clicked.
+ */
+function addLandmark(map, lat, lng, title, URL) {
+  const marker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
+    map: map,
+    animation: google.maps.Animation.DROP});
+
+  const infoWindow = new google.maps.InfoWindow({content: title});
+  marker.addListener('click', async () => {
+    infoWindow.open(map, marker);
+    await htmlInject(URL, 'description-container');
```

Comment:
This design is cool because it will only load the description HTML when the user clicks on the map (saving on the amount of data you need to load when the page first opens).  Just be aware that if you server or network connection is slow, there might be a noticeable lag between when you click and when you get the data.  If that's a problem you could possibly consider a different design where all the landmark HTML or info is included in either the JS or HTML when the page first loads.

No changes needed now, just thought I'd mention it.

Reply:
Ok, that makes sense. I'll keep this in mind in the future, and with other projects.